src/couch_replicator/src/couch_replicator_utils.erl

```diff
@@ -276,6 +278,88 @@ seq_encode(Seq) ->
     % object. We are being maximally compatible here.
     ?JSON_ENCODE(Seq).
 
+%% Log uses of http protocol and uses of https protocol where verify_peer
+%% would fail.
+log_security_warnings(#rep{} = Rep) ->
```

Comment:
A minor nit: `security` seems a bit too general, we're not checking all possible security issues, just the endpoint protocols.

Reply:
suggestions welcome, I don't find it too general but I don't feel strongly either way.

Reply:
Perhaps that's closer the valid_endpoint_protocols/valid_proxy_protocols https://docs.couchdb.org/en/stable/config/replicator.html#replicator/valid_endpoint_protocols

Same prefix with a `_log` suffix like `valid_endpoint_protocols_log` it would take the same values as (list of allowed schemas)? When the option with `_log` suffix is used we'd only log a warning when it's used without the log the jobs will be rejected or become `failed`.